Sources/Event Logging (Swift)/TrackEventPersistenceService.swift

```diff
@@ -0,0 +1,82 @@
+#if SWIFT_PACKAGE
+import AutomatticTracksModel
+import AutomatticTracksModelObjC
+#endif
+import CoreData
+
+/// Should eventually replace objC class `TracksEventPersistenceService`.  Unfortunately
+/// Swift Packages do not support mixed source classes so we'll instead need to compose.
+///
+@objc
+public class TrackEventPersistenceService: NSObject {
+    
+    private let managedObjectContext: NSManagedObjectContext
+    
+    @objc
+    public init(managedObjectContext: NSManagedObjectContext) {
+        self.managedObjectContext = managedObjectContext
+        
+        super.init()
+    }
+    
+    /// Increments the retry count for the specified events in batches of
+    @objc
+    public func incrementRetryCountForEvents(_ tracksEvents: [TracksEvent], onComplete completion: ((Error?) -> Void)?) {
+        let uuidStrings = tracksEvents.map { event in
+            event.uuid.uuidString
+        }
+        
+        let batchSize = 500
+        
+        for startIndex in stride(from: 0, to: uuidStrings.count, by: batchSize) {
+            let isLastBatch = startIndex + batchSize >= uuidStrings.count
+            
+            managedObjectContext.perform {
+                let results: [TracksEventCoreData]
+                let count = min(uuidStrings.count - startIndex, batchSize)
+                let uuidStringsBatch = Array(uuidStrings[startIndex ..< startIndex + count])
+                
+                do {
+                    results = try self.findTrackEventCoreData(uuidStrings: uuidStringsBatch)
+                } catch {
+                    TracksLogError("Error while finding track events: \(String(describing: error))")
+                    
+                    if isLastBatch {
+                        completion?(error)
+                    }
+                    
+                    return
+                }
+                
+                if results.count != count {
+                    TracksLogError("Not all provided events were found in the persistence layer. This signals a possible logical error in the tracking, persistence and retry-count-incrementing code.  Please review.")
```

Comment:
Partial errors shouldn't halt the process, but we'll probably want them logged just in case.  This type of error shouldn't really happen though, so it may be overkill to even try to log this.